best-practices/gke-batch-refarch/cloudbuild-create.yaml

```diff
@@ -61,61 +57,135 @@ steps:
+          terraform init -force-copy -migrate-state -no-color && \
+          rm -rf state || exit 1
+        fi
+
+        cd ..
+        
+        # Networking
+        cd networking && \
+        terraform init -no-color && \
+        terraform plan -input=false -no-color -out=tfplan ${tf_variables} && \
+        terraform apply -input=false -no-color tfplan && \
+        rm tfplan || exit 1
+        cd ..
+
+        # Container cluster
+        cd container_cluster && \
+        terraform init -no-color && \
+        terraform plan -input=false -no-color -out=tfplan ${tf_variables} && \
+        terraform apply -input=false -no-color tfplan && \
+        rm tfplan  || exit 1
+        cd ..
+
+        # Workloads
+        cd workloads && \
+        terraform init -no-color && \
+        terraform plan -input=false -no-color -out=tfplan ${tf_variables} && \
+        terraform apply -input=false -no-color tfplan && \
+        rm tfplan  || exit 1
+        cd ..
+
+        # Get cluster credentials
+        gcloud container clusters get-credentials ${TF_VAR_cluster_name_prefix}-${TF_VAR_environment_name} --region=${_REGION} --project=${PROJECT_ID}
+
   - name: "${_REGION}-docker.pkg.dev/${PROJECT_ID}/tutorial-installer/installer"
-    id: "Install GPU drivers"
+    id: "Configure GKE for Batch Reference Architecture"
+    dir: "gke-batch-refarch/gke"
     entrypoint: "ash"
-    waitFor: ["Setup GKE"]
+    waitFor:
+      ["Build Installer Image", "Setup GKE"]
     args:
       - "-xe"
       - "-c"
       - |
-        gcloud container clusters get-credentials gke-batch-refarch --region=${_REGION} --project=${PROJECT_ID} && \
-        kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/master/nvidia-driver-installer/cos/daemonset-preloaded-latest.yaml
+        terraform init -backend-config="bucket=${PROJECT_ID}-batch-dev-terraform" -var="project_id=${PROJECT_ID}" -var="region=${_REGION}" -var="zone=${_ZONE}" -no-color && \
+        terraform plan -out terraform.tfplan -var="project_id=${PROJECT_ID}" -var="region=${_REGION}" -var="zone=${_ZONE}" -no-color && \
+        terraform apply -input=false -lock=false terraform.tfplan -no-color && \
+        rm terraform.tfplan
+
   - name: "${_REGION}-docker.pkg.dev/${PROJECT_ID}/tutorial-installer/installer"
     id: "Install Kueue and JobSet"
     entrypoint: "ash"
-    waitFor: ["Setup GKE"]
+    waitFor: 
+      ["Setup GKE", "Configure GKE for Batch Reference Architecture"]
     args:
       - "-xe"
       - "-c"
       - |
-        gcloud container clusters get-credentials gke-batch-refarch --region=${_REGION} --project=${PROJECT_ID} && \
-        kubectl apply --server-side -f https://github.com/kubernetes-sigs/jobset/releases/download/v0.2.3/manifests.yaml && \
+        kubectl apply --server-side -f https://github.com/kubernetes-sigs/jobset/releases/download/v0.6.0/manifests.yaml && \
```

Comment:
Is it possible to use a variable for jobset version ? It'll help for future release

Reply:
We might consider moving the jobset out to the base platform where it can be a variable.